src/docs/development/platform-integration/platform-views.md

```diff
@@ -0,0 +1,248 @@
+    }
+
+    @NonNull
+    @Override
+    public View getView() {
+        return textView;
+    }
+
+    @Override
+    public void dispose() {}
+}
+```
+
+In your factory class, for example `NativeViewFactory.java`:
+
+`NativeViewFactory.java`
+
+```java
+package dev.flutter.example;
+
+import android.content.Context;
+import android.view.View;
+import androidx.annotation.Nullable;
+import androidx.annotation.NonNull;
+import io.flutter.plugin.common.BinaryMessenger;
+import io.flutter.plugin.common.StandardMessageCodec;
+import io.flutter.plugin.platform.PlatformView;
+import io.flutter.plugin.platform.PlatformViewFactory;
+import java.util.Map;
+
+class NativeViewFactory extends PlatformViewFactory {
+  @NonNull private final BinaryMessenger messenger;
+  @NonNull private final View containerView;
+
+  WebViewFactory(@NonNull BinaryMessenger messenger, @NonNull View containerView) {
+    super(StandardMessageCodec.INSTANCE);
+    this.messenger = messenger;
+    this.containerView = containerView;
+  }
+
+  @NonNull
+  @Override
+  public PlatformView create(@NonNull Context context, int id, @Nullable Object args) {
+    final Map<String, Object> creationParams = (Map<String, Object>) args;
+    return new NativeView(context, id, creationParams);
+  }
+}
+```
+
+In your main activity file, for example `MainActivity.java`:
+
+```java
+package dev.flutter.example;
+
+import androidx.annotation.NonNull;
+import io.flutter.embedding.android.FlutterActivity;
+import io.flutter.embedding.engine.FlutterEngine;
+
+public class MainActivity extends FlutterActivity {
+    @Override
+    public void configureFlutterEngine(@NonNull FlutterEngine flutterEngine) {
+        flutterEngine
+            .getPlatformViewsController()
+            .getRegistry()
+            .registerViewFactory("hybrid-view-type", new NativeViewFactory());
+    }
+}
+```
+
+For more information, see the API docs for:
+
+* [`PlatformViewRegistry`][]
+* [`PlatformViewFactory`][]
+* [`PlatformView`][]
+
+
+[`PlatformViewFactory`]: {{site.api}}/javadoc/io/flutter/plugin/platform/PlatformViewFactory.html
+[`PlatformViewRegistry`]: {{site.api}}/javadoc/io/flutter/plugin/platform/PlatformViewRegistry.html
+[`PlatformView`]: {{site.api}}/javadoc/io/flutter/plugin/platform/PlatformView.html
+
+Finally, modify your `build.gradle` file to require
+version 19 (or later) of the Android API.
+
+```gradle
+android {
+    defaultConfig {
+        minSdkVersion 19
+    }
+}
+```
+
+### Performance
+
+Hybrid composition on Android works best on Android 10 or above.
+On lower versions of Android, the Flutter UI can appear slower
+while the native view is rendered on the screen.
+This overhead is due to the synchronization of Flutter frames to the
+Android view system.
+
+To mitigate this issue, we recommend that you avoid displaying
+the native view while an animation is taking place in Dart.
+For example, you could use a placeholder such as capturing a
```

Comment:
It's not completely trivial for someone reading this to know how to do it, if we suggest it maybe refer to an example?

Reply:
Agreed! @blasten has indicated that he will add examples when I land this PR.